Feature/sgs stats (#2286)
ExtremeFLOW/neko: doc/pages/user-guide/statistics-guide.md

e_{vs} &= \frac{\partial v}{\partial x}  \frac{\partial s}{\partial x} + \frac{\partial v}{\partial y}\frac{\partial s}{\partial y}+ \frac{\partial v}{\partial z}\frac{\partial s}{\partial z} \\\\
e_{ws} &= \frac{\partial w}{\partial x}  \frac{\partial s}{\partial x} + \frac{\partial w}{\partial y}\frac{\partial s}{\partial y}+ \frac{\partial w}{\partial z}\frac{\partial s}{\partial z} \\\\
\end{aligned}
$$

# Fluid Subgrid-Scale (SGS) Statistics

> In the fluid SGS statistics in Neko, subgrid-scale (SGS) contribution to the anisotropic part of the Reynolds stresses are computed, in a similar fashion to the fluid statistics. In total, 7 statical quantities are computed.

Reviewer: ```suggestion
In the fluid SGS statistics in Neko, subgrid-scale (SGS) contribution to the anisotropic part of the Reynolds stresses are computed, in a similar fashion to the fluid statistics. In total, 7 statistical quantities are computed.
```

| `output_filename`        | User-specified filename to store output in.                       | filename  |  fluid_sgs_statsX*        |

\*The name of the written statistics file will by default be `scalar_sgs_statsX0.f0000X,..., scalar_sgs_statsX0.f0000Y` where X is the number of the first outputted statistic of the current run.

In addition, one can specify the usual controls for the output, in the same manner as for fluid statistics. For example, if one wants to compute only the basic statistics and sample the fields every 4 time steps and compute and output batches every 20 time units and have an initial transient of 60 time units the following would work if the eddy diffusivity field is registered:

~~~~~~~~~~~~~~~{.json}
"simulation_components":
  [
    {
      "type": "scalar_sgs_stats",
      "field": "temperature",
      "alphat": {
        "nut_dependency": false,
        "alphat_field": "temperature_alphat"
      },
      "compute_control": "tsteps",
      "compute_value": 4,
      "output_control": "simulationtime",
      "output_value": 20,
      "start_time": 60.0,
      "avg_direction":"xz",
    }
  ]
~~~~~~~~~~~~~~~

> Otherwise, if one does not use model for the scalar eddy diffusivity but relates it with the eddy viscosity, the following would work is one specify a turbulent Prandtl number to be for instance 0.7 and the output will give the eddy diffusivity field as the edy viscosity field divided by the turbulent Prandtl number:

Reviewer: ```suggestion
Otherwise, if one does not use a model for the scalar eddy diffusivity but relates it to the eddy viscosity, one needs to specify the turbulent Prandtl number, for instance, 0.7. The output will give the eddy diffusivity field as the eddy viscosity field divided by the chosen turbulent Prandtl number:
```
